state-chain/chains/src/btc/api.rs

```diff
@@ -146,6 +147,34 @@ impl<E: ReplayProtectionProvider<Bitcoin>> ExecutexSwapAndCall<Bitcoin> for Bitc
 	}
 }
 
+impl<E: ReplayProtectionProvider<Bitcoin>> RejectCall<Bitcoin> for BitcoinApi<E>
+where
+	E: ChainEnvironment<UtxoSelectionType, SelectedUtxosAndChangeAmount>
+		+ ChainEnvironment<(), AggKey>,
+{
+	type DepositDetails = <Bitcoin as Chain>::DepositDetails;
+	fn new_unsigned(
+		deposit_details: Self::DepositDetails,
+		refund_address: ForeignChainAddress,
+		amount: <Bitcoin as Chain>::ChainAmount,
+	) -> Result<Self, RejectError> {
+		let agg_key = <E as ChainEnvironment<(), AggKey>>::lookup(()).ok_or(RejectError::Other)?;
+		let script_pubkey = match refund_address {
+			ForeignChainAddress::Btc(script_pubkey) => script_pubkey,
+			_ => return Err(RejectError::Other),
+		};
+		Ok(Self::NoChangeTransfer(BitcoinTransaction::create_new_unsigned(
+			&agg_key,
+			vec![Utxo {
+				amount,
```

Comment:
The input amount has to be different from the one specified in the output, otherwise the gas fee is 0, correct? Looks like `amount` passed here already excludes egress fee, but I think the input amount should be the full amount of the UTXO that we want to refund.

Reply:
Bitcoin deposit details will need to be changed to Utxo.